catalog/ui/search-ui/standard/src/main/java/org/codice/ddf/ui/searchui/standard/properties/ConfigurationStore.java

```diff
@@ -266,21 +262,71 @@ public void setTerrainProvider(String terrainProvider) {
     }
 
     private void setProxiesForImagery(List<String> imageryProviders) {
+        List<Map<String, Object>> newImageryProviders = imageryProviders.stream()
+                .map(this::getProviderConfigMap)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        List<Map<String, Object>> imageryProvidersToStop = new ArrayList<>();
+        List<Map<String, Object>> imageryProvidersToStart = new ArrayList<>();
+
+        findDifferences(proxiedImageryProviders, newImageryProviders, imageryProvidersToStart);
+        findDifferences(newImageryProviders, proxiedImageryProviders, imageryProvidersToStop);
+
+        List<String> proxiesToStop = imageryProvidersToStop.stream()
+                .map(provider -> urlToProxyMap.get(provider))
+                .collect(Collectors.toList());
+
+        stopImageryEndpoints(proxiesToStop);
+        startImageryEndpoints(imageryProvidersToStart);
+        proxiedImageryProviders = newImageryProviders;
+    }
+
+    private void findDifferences(List<Map<String, Object>> innerList,
```

Comment:
I'm confused by this method a little bit, actually. Testing the equality of the two maps will compare their entrySets, but I'm not clear how the value mapped to the key `URL` is going to be the same when comparing the `newImageryProvider` list of maps against ones that have already been processed. I assume/hope I'm just missing something.